### ISSUE TITLE
[Feature Request] Load BSA files loaded by MO.

### ISSUE THREAD

**Issue by tony971** [TES5Edit/TES5Edit#334]
Okay I did some weird stuff with [Optimized Vanilla Textures](http://www.nexusmods.com/skyrim/mods/57353/?) again that needs new xEdit functionality. So we all know that ModOrganizer has the ability to load BSA files without a plugin or ini edit. In previous versions, OVT took advantage of this by not even supplying a plugin to MO users. Well, this is still the case with the newest version, but with an extra wrench.

I took all of the textures from Skyrim - Textures.bsa and merged them with High Res DLC files that would override them. This would have generated a BSA file larger than 2gb so I split it into halves - "Skyrim - HD01" and "Skyrim - HD02." Mod Organizer loads these both without fail.

When TES5Edit is run through MO, it doesn't load these files. Instead, it checks the traditional locations of ini files and plugins to know what to load. Here's where it starts being necessary to load the BSA files from MO: I realized that if I renamed "Skyrim - HD01.bsa" to "Skyrim - Textures.bsa" that MO would load it instead of the vanilla BSA. This saves resources since the original textures aren't being loaded anymore. The problem is that Skyrim - HD01.bsa only has half of the game's textures and TES5Edit (run through MO) only loads that half. So the textures in Skyrim - HD02.bsa are never loaded into TES5Edit, taking away half of the game's textures.

Programs that rely on xEdit's ability to load the game's resources, like DynDOLOD, now fail to load half the game's textures and don't complete properly.

**Comment by GrantSP** [comment]
Actually I don't think this will be needed. Since MO 2 will be dropping the whole "manage BSAs without plugins" concept entirely, and shifting to a pure plugin+BSA load mechanism, this will only affect MO version 1 users and it's a simple thing to just be sure to not use this method.
On top of that I don't really think it is in the realms of capability to have xEdit check the "fake" VFS MO employs for those BSAs, or for that matter to add in a specific set of functions to handle only one mod manager's peculiarities.
Keep in mind I'm an ardant supporter of MO and one of the support team so I feel I have a slightly skewed view of things.

**Comment by tony971** [comment]
And what about the mods that include dummy plugins to load BSA files? MOST* MO users disable these plugins. Are their BSA files not getting loaded into TES5Edit, either?

**Comment by GrantSP** [comment]
Good question. I honestly don't know. But "most" users will be using xEdit to just check for conflicts in plugin records so I'm guessing it won't affect them to the same severity as this.
Still it is a truly fascinating find you made and the consequences will need to be fleshed out to see what we need to do.

**Comment by EssArrBee** [comment]
MO users need to be aware that the dummy plugin feature is not for utilities. It's a feature of MO loading game executables. If you have something that needs to see the plugins then they must be enabled in the load order.

**Comment by tony971** [comment]
That's not something immediately obvious and rarely talked about.

**Comment by zilav** [comment]
As I mentioned many times - it is the duty of virtualization system to support applications it is providing virtualization features for, not the other way around. Have you ever seen a line like this in changelog of any application "New in version 1.2.3: Supports running under VMWare Player"?

The whole BSA management feature of MO was poorly implemented, and in general a mistake that caused a lot of compatibility issues, especially for 3rd party tools. Great that it is finally being removed, less problems for end users, less hassle with documentation parts designated for MO users exclusively. Honestly it was ridiculous that when I wrote some description, I had to add lines about MO to workaround it's "features" and bugs to make tools work with it.
